brave/src/main/java/brave/propagation/ExtraFieldPropagation.java

```diff
@@ -272,15 +275,39 @@ public static void set(TraceContext context, String name, String value) {
     PropagationFields.put(context, lowercase(name), value, Extra.class);
   }
 
-  public static final class Factory extends Propagation.Factory {
+  public static class Factory extends Propagation.Factory {
     final Propagation.Factory delegate;
+
+    Factory(Propagation.Factory delegate) {
+      this.delegate = delegate;
+    }
+
+    @Override public <K> ExtraFieldPropagation<K> create(Propagation.KeyFactory<K> keyFactory) {
+      return new ExtraFieldPropagation<>(delegate, keyFactory);
+    }
+
+    @Override public boolean supportsJoin() {
+      return delegate.supportsJoin();
+    }
+
+    @Override public boolean requires128BitTraceId() {
+      return delegate.requires128BitTraceId();
+    }
+
+    @Override public TraceContext decorate(TraceContext context) {
+      return delegate.decorate(context);
+    }
+  }
+
+  public static final class RealFactory extends Factory {
```

Comment:
I believe we can keep this class package-private

Reply:
good catch!